room_allocation.py

```diff
@@ -104,22 +120,246 @@ def get_room_by_name(self, room_name):
+        Print list of people not in a room
+        '''
+        if len(self.unallocated_people) < 1:
+            return "No unallocated people exist."
+
+        all_data = '\n'.join(self.unallocated_people.keys())
+
+        if filename is None:
+            return all_data
+
+        directory = "test_files/"
+        filepath = directory + filename + '.txt'
+        if filename and self.is_valid_filename(filename):
+            f = open(filepath, "w")
+            f.write(all_data)
+            f.close()
+            return "Wrote to " + filepath + " successfully."
+        else:
+            return "Invalid filename."
+
+    def save_state(self, db='amity'):
+        '''
+        Persists the data of the app in database and stores it in the 
+        /databases folder
+        args:
+            db - sqlite database to save to (name to give databases)
+
+        '''
+
+        all_rows = []  # both people and room rows
+        for room_obj in self.rooms.values():
+            row = room_obj.to_db_row()
+            row.people_in_room = room_obj.people_in_room_to_rows()
+            all_rows.append(row)
+            all_rows += row.people_in_room
+
+        for person_obj in self.unallocated_people.values():
+            row = person_obj.to_db_row()
+            all_rows.append(row)
+
+        directory = "databases/"
+        filepath = directory + db + '.db'
+        if self.is_valid_filename(db):
+            with open(filepath, 'wb') as f:
+                pass
+            db = connect_db(filepath)
+            db.add_all(all_rows)
+            db.commit()
+            return "Data was successfuly saved to " + filepath
+
+    def load_state(self, dbpath):
+        '''
+        Loads data from a database into Amity.
+        args:
+            dbpath - sqlite database to load from
+        '''
+        if not os.path.exists(dbpath):
+            return "Invalid file path."
+
+        db = connect_db(dbpath)
+        rooms = db.query(ModelRoom).all()
+        people = db.query(ModelPerson).all()
+
+        for room in rooms:
+            if not room.name in self.rooms:
+                self.rooms[room.name] = room.to_obj()
+                people_in_room = room.people_in_room_to_objs()
+                self.rooms[room.name].people_in_room = people_in_room
+            else:
+                return "Room already exists."
+
+        for person in people:
+            if not person.name in self.all_people:
+                self.all_people[person.name] = person.to_obj()
+
+            if not person.room_id:
+                self.unallocated_people[person.name] = person.to_obj()
+
+        return "Load successful!"
+
 
 class Room(object):
     def __init__(self, name):
         '''
         Creates room with given name
         '''
         self.name = name
-        self.people_in_room = []
+        self.people_in_room = {}
+        self.max_capacity = None
+
+    def is_full(self):
+        if len(self.people_in_room) >= self.max_capacity:
```

Comment:
This should be == not >= because that means an lspace can have 10 people based on '>`